scr/main/java/bittercode/model/Address.java

```diff
@@ -0,0 +1,71 @@
+package com.bittercode.model;
+
+import java.io.Serializable;
+
+public class Address implements Serializable {
+
+    private String addressLine1;
+    private String addressLine2;
+    private String city;
+    private String state;
+    private String country;
+    private long pinCode;
+    private String phone;
+
+    public String getAddressLine1() {
+        return addressLine1;
+    }
+
+    public void setAddressLine1(String addressLine1) {
+        this.addressLine1 = addressLine1;
+    }
+
+    public String getAddressLine2() {
+        return addressLine2;
+    }
+
+    public void setAddressLine2(String addressLine2) {
+        this.addressLine2 = addressLine2;
+    }
+
+    public String getCity() {
+        return city;
+    }
+
+    public void setCity(String city) {
+        this.city = city;
+    }
+
+    public String getState() {
+        return state;
+    }
+
+    public void setState(String state) {
+        this.state = state;
+    }
+
+    public String getCountry() {
+        return country;
+    }
+
+    public void setCountry(String country) {
+        this.country = country;
+    }
+
+    public long getPinCode() {
+        return pinCode;
+    }
+
+    public void setPinCode(long pinCode) {
+        this.pinCode = pinCode;
+    }
+
+    public String getPhone() {
+        return phone;
+    }
+
+    public void setPhone(String phone) {
+        this.phone = phone;
+    }
+
+}
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider adding address field validation</b></div>
<div id="fix">

Consider adding validation for address fields. For example, `pinCode` and `phone` should be validated for proper formats, and address lines should check for reasonable length limits.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -19,6 +19,9 @@
     public void setAddressLine1(String addressLine1) {
 +        if (addressLine1 == null || addressLine1.trim().isEmpty()) {
 +            throw new IllegalArgumentException("Address line 1 cannot be null or empty");
 +        }
         this.addressLine1 = addressLine1;
     }
```
</div>
</details>
</div>

<div id="suggestion">
<div id="issue"><b>Consider string type for postal codes</b></div>
<div id="fix">

Consider using `String` instead of `long` for `pinCode` to preserve leading zeros and handle international postal codes that may contain letters.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -12,1 +12,1 @@
 -    private long pinCode;
 +    private String pinCode;
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2517477681>#c23dbc</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged